### PR TITLE
Rspec tests

### PR DESCRIPTION
**Pull request by JohannRodriguez**
- Created Rspec to test the code.
- Added different expectations for each method.
- Main code changed to accept more cases

### CODE REVIEW

File: spec/enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  end
+
+  describe '#my_each_with_index' do
+    it 'iterates over each item and returns item with its index' do
+      expect(test_array_numbers.my_each_with_index { |n, i| [n, i] }).to eql([11, 0, 2, 1, 3, 2, 56, 3])
+      expect(test_array_numbers.my_each_with_index).to be_kind_of(Enumerator)
+    end
+  end
+
+  describe '#my_select' do
```
Comment (mosesogwo) [inline spec/enumerables_spec.rb:23]: Add test for 'when no block is given' to my_select

File: spec/enumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      expect(test_array_numbers.my_select(&:even?)).to eql([2, 56])
+      expect(test_range.my_select(&:odd?)).to eql([1, 3])
+    end
+  end
+
+  describe '#my_all?' do
+    it 'returns true if all items in an array pass a test' do
+      expect(test_array_string.my_all? { |word| word.length >= 3 }).to eql(true)
+    end
+    it 'returns true if all numbers in an array pass a test' do
```
Comment (mosesogwo) [inline spec/enumerables_spec.rb:34]: This is not  a correct description of the test

File: spec/enumerables_spec.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  end
+
+  describe '#my_select' do
+    it 'returns an array of selected items that pass a test' do
+      expect(test_array_numbers.my_select(&:even?)).to eql([2, 56])
+      expect(test_range.my_select(&:odd?)).to eql([1, 3])
+    end
+  end
+
+  describe '#my_all?' do
```
Comment (mosesogwo) [inline spec/enumerables_spec.rb:30]: Add test for when no block is given to my_all?

File: spec/enumerables_spec.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    it 'returns true if all numbers in an array pass a test' do
+      expect(test_array_numbers.my_all? { |i| i > 2 }).to eql(false)
+    end
+    it 'returns true if all numbers of a range pass a test' do
+      expect(test_range.my_all? { |i| i < 5 }).to eql(true)
+    end
+  end
+
+  describe '#my_any?' do
+    it 'returns true if any item in an array pass a given test' do
```
Comment (mosesogwo) [inline spec/enumerables_spec.rb:43]: Here, no block is given. Add another test for when there is a test block.

File: spec/enumerables_spec.rb [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    end
+    it 'returns false if no value is given' do
+      expect([].my_any?).to eql(false)
+    end
+    it 'return true if any item of the array is equal to argument' do
+      expect(test_array_numbers.my_any?(11)).to eql(true)
+    end
+  end
+
+  describe '#my_none?' do
```
Comment (mosesogwo) [inline spec/enumerables_spec.rb:54]: Add test for when no block or argument is given

File: spec/enumerables_spec.rb [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  end
+
+  describe '#my_count' do
+    it 'returns the number of items that return true for a given test' do
+      expect(test_array_numbers.my_count { |x| x > 10 }).to eql(2)
+      expect(test_array_string.my_count('ant')).to eql(1)
+    end
+  end
+
+  describe '#my_map' do
```
Comment (mosesogwo) [inline spec/enumerables_spec.rb:70]: Add test for when no block is given

File: spec/enumerables_spec.rb [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    it 'returns an array of results of running a block with each item' do
+      expect(test_range.my_map { |i| i * i }).to eql([1, 4, 9, 16])
+      expect(test_array_numbers.my_map { |i| i * 2 }).to eql([22, 4, 6, 112])
+      expect(test_array_string.my_map { |i| i.length }).to eql([3, 4, 3])
+    end
+  end
+
+  describe '#my_inject' do
```
Comment (mosesogwo) [inline spec/enumerables_spec.rb:78]: Add test for when a starting value is passed as an argument.

**Comment by pelzolga123** [comment]
## Great work, Congratulation :+1: Project Approved.
You did all changes required previously. Go ahead and merge to develop.
_Goodluck in your next project_ *Happy coding*

**Pull request merged**